tt_metal/jit_build/build.cpp

```diff
@@ -491,16 +494,55 @@ void JitBuildState::elf_to_hex8(const string& log_file, const string& out_dir) c
     }
 }
 
-void JitBuildState::hex8_to_hex32(const string& log_file, const string& out_dir) const
-{
-    string cmd;
-    cmd = "cd " + out_dir + " && ";
-    cmd += env_.hex8tohex32_ + this->target_name_ + ".hex.tmp " + this->target_name_ + ".hex";
-
-    log_debug(tt::LogBuildKernels, "    hex8tohex32 cmd: {}", cmd);
-    if (!tt::utils::run_command(cmd, log_file, false)) {
-        build_failure(this->target_name_, "hex8tohex32.py", cmd, log_file);
+void hex8_to_hex32(const string& log_file, const string& out_dir, const string& target_name){
+    ZoneScoped;
+    auto write_data = [](std::ofstream& outf, std::vector<long>& data, long& ptr){
```

Comment:
we don't really use long type in our code. we prefer using types from cstdint (examples uint32_t)

Reply:
changed